src/js/providers/providers-register.ts

```diff
@@ -3,7 +3,7 @@ import ProvidersSupported from 'providers/providers-supported';
 import DefaultProvider from 'providers/default';
 import { find, matches, isFunction, defaults } from 'utils/underscore';
 
-export default function registerProvider(provider) {
+export default function registerProvider(provider: DefaultProvider): void {
```

Comment:
once the other providers are typed this will need to be a union type, thoughts on creating a `type ProviderUnion` in `generic.type.ts` as a placeholder (currently could just be `type ProviderUnion = DefaultProvider`)

Reply:
only html5 provider is defined this project. The type should be a base class or interface. Custom providers are supported too.